core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/ItemSelectionInputCheckerService.js

```diff
@@ -30,17 +30,26 @@ oppia.factory('ItemSelectionInputCheckerService', [
         // - Each answer group has exactly one rule (for an "Equals" rule). The
         //   argument for each of these rules should match a choice in
         //   customizationArgs.
-        // - Each answer group corresponds to a different choice.
+        // - Each answer group corresponds to a different set of choices.
         var numChoices = customizationArgs.choices.value.length;
         var coveredChoices = [];
         for (var i = 0; i < answerGroups.length; i++) {
           var rules = answerGroups[i].rules;
           if (rules.length !== 1 ||
-              rules[0].type !== 'Equals' || rules[0].inputs.x >= numChoices) {
+              rules[0].type !== 'Equals' || 
+              rules[0].inputs.x.length > numChoices) {
             return false;
           }
-          if (coveredChoices.indexOf(rules[0].inputs.x) !== -1) {
-            return false;
+          for (var j = 0; j < coveredChoices.length; j++) {
+            var matched = 0;
+            for (var k = 0; k < rules[0].inputs.x.length; k++) {
+              if (coveredChoices[j].indexOf(rules[0].inputs.x[k]) !== 1) {
```

Comment:
I don't follow most of the logic here, I'm afraid. Could you explain what's going on?

Reply:
For this, I was making sure if any of the arrays within `coveredChoices` matched the array of `rules[0].inputs.x` (with same elements, but in any order). `indexOf` doesn't seem to work on arrays in this case.

Reply:
Not sure what you mean by "doesn't seem to work", but I noticed you are comparing to 1 instead of -1. I'm not sure why...

Reply:
`indexOf` is returning `-1` even though there are arrays that match (I assume that it is because it only works on more "primitives", but could not find type restrictions in the docs). Yup, it is supposed to be `-1`

Reply:
Hm, ok. Just as a check, you might want to look at extensions/interactions/ItemSelectionInput/ItemSelectionInputValidationService.js. It's possible you might be able to copy/reuse some of the ideas there.

Reply:
It's close, but I did borrow some things from over there.